.github/workflows/build-and-test.yaml

```diff
@@ -11,13 +11,21 @@ jobs:
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-python@v5
+      - name: Install Poetry
```

Comment:
Looks like this PR for some reason contains other changes from master branch, Can you please fix that since the PR is already huge and it is very hard to understand which change is relevant and which isn't. 

Thanks

Reply:
I think this was because the branch was not up to date.